dj/sql/parsing/ast.py

```diff
@@ -679,6 +817,17 @@ class Query(Expression):
     select: "Select"
     ctes: List[Alias["Select"]] = field(default_factory=list)
 
+    def to_select(self) -> Select:
+        """compile ctes into the select and return the select
+
+        Note: This destroys the structure of the query which cannot be undone
```

Comment:
This message about the mutation seems a bit buried. Would it be reasonable to have another layer that does a deepcopy instead of changing the structure of the query, and then rename this method to be `_to_select`, to indicate that it's not meant for external use.

Reply:
I can look into this again. I had some trouble getting a deepcopy to work as I expected and since this was meant purely as a utility to ditch ctes, I just tabled the copy. The different stages modify in-place anyways. Not sure what uses might be for people to do something like compile then put that ast aside, then build and then have both asts sitting around.

Reply:
Thinking now, `_to_select` makes the most sense. Thanks!

Reply:
done!